### ISSUE TITLE
Lookup error caused by field prefixed with "@"

### ISSUE THREAD

**Issue by spencergilbert** [timberio/vector#4273]
### Vector Version

```
nightly from oct 2
```

### Vector Configuration File

```toml
[transforms.add_fields]
	type = "add_fields"
	inputs = ["input"]
	overwrite = false
	fields."@timestamp" = "{{ timestamp }}"
```

### Expected Behavior

Happily parse a field starting with "@"

### Actual Behavior
```
Oct 02 20:55:18.418  INFO vector::app: Log level "info" is enabled.
Oct 02 20:55:18.419  INFO vector::app: Loading configs. path=["/etc/vector/vector.toml"]
Oct 02 20:55:18.448 ERROR vector::topology: Configuration error: Transform "add_fields":  --> 1:1
  |
1 | @timestamp
  | ^---
  |
  = expected lookup
```

I discussed with @Hoverbear briefly and should have a PR in shortly.

**Comment by JeanMertz** [comment]
Re-assigning myself, since I've already started working on this locally.

**Comment by gustavoamigo** [comment]
@JeanMertz just to let you know that I had a similar problema with the nightly build, but with the rename and remove transformations. 
See https://github.com/timberio/vector/issues/4582#issuecomment-711269897

**Comment by binarylogic** [comment]
@JeanMertz was this resolved through a PR?

**Comment by JeanMertz** [comment]
@binarylogic ah, I dragged this in Zenhub without realizing this got closed.

But yes and no, this got "resolved" because #4804 got merged, which reverts the Lookup changes. We'll make sure this works once we re-introduce this before the next release, but there's no longer a "bug" for now.

It's fine leaving this closed, as we're aware of the fact that we need to make sure Lookup is backward compatible.
